8Knot/pages/repo_overview/visualizations/repo_general_info.py

```diff
@@ -68,18 +75,38 @@ def repo_general_info(repo):
     # test if there is data
     if df_repo_files.empty and df_repo_info.empty and df_releases.empty:
         logging.warning(f"{VIZ_ID} - NO DATA AVAILABLE")
-        return dbc.Table.from_dataframe(pd.DataFrame(), striped=True, bordered=True, hover=True)
+        return dbc.Table.from_dataframe(pd.DataFrame(), striped=True, bordered=True, hover=True), dbc.Label("No data")
 
-    df = process_data(df_repo_files, df_repo_info, df_releases)
+    df, last_updated = process_data(df_repo_files, df_repo_info, df_releases)
 
     table = dbc.Table.from_dataframe(df, striped=True, bordered=True, hover=True)
 
     logging.warning(f"{VIZ_ID} - END - {time.perf_counter() - start}")
-    return table
+    return table, last_updated
 
 
 def process_data(df_repo_files, df_repo_info, df_releases):
 
+    # get all values from the data_collection_date column
+    # updated_times_repo_files = pd.to_datetime(df_repo_files["data_collection_date"])
+    updated_times_repo_info = pd.to_datetime(df_repo_info["data_collection_date"])
```

Comment:
lets do this date based of the df_repo_info["data_collection_date"]

Reply:
Thats currently what is being done, I just have extra steps commented out there in case you wanted to do something else.


Would you like me to make any changes here?

Reply:
Seems like this could be done in 1 to 2 lines? If not could you explain why what you have is needed?

Reply:
ah yeah most of it was stuff i left for debugging/flexibility in case you wanted something else (i.e. using all the dataframes to source the time).

I simplified some of the date code (and removed the try catch) in the commits i just made)